pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+    """Start a content upload.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + _CONTENT_UPLOADS_PATH,
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _upload_file(server_config, href, content, responses=None):
+    """Upload a file and return the server's response.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to an open upload request.
+    :param content: The content of the file to upload.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.put(
+        server_config.base_url + href + '0/',
+        data=content,
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _import_rpm_to_repo(server_config, upload_id, href, responses=None):
+    """Import an RPM from an upload into a repository.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param upload_id: A string. The ID of an upload request.
+    :param href: A string. The path to an RPM repository.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'actions/import_upload/',
+        json={'unit_key': {}, 'unit_type_id': 'rpm', 'upload_id': upload_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _copy_repo(server_config, source_repo_id, href, responses=None):
+    """Copy content from one repository to another.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param source_repo_id: A string. The ID of the source repository.
+    :param href: A string. The path to the repository on which the association
+        action is being performed. Content is copied to this repo.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return _handle_response(requests.post(
+        server_config.base_url + href + 'actions/associate/',
+        json={'source_repo_id': source_repo_id},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _add_yum_distributor(server_config, href, responses=None):
```

Comment:
Cool. If it's just the `distributor_type_id`, then something like this could be written:

```python
def add_distributor(server_config, href, distributor_type_id, responses=None):
    pass
```